src/vizdoom/src/sound/oalsound.cpp

```diff
@@ -1,2017 +1,2293 @@
+        // EFX function pointers
+        LOAD_FUNC(alGenEffects);
+        LOAD_FUNC(alDeleteEffects);
+        LOAD_FUNC(alIsEffect);
+        LOAD_FUNC(alEffecti);
+        LOAD_FUNC(alEffectiv);
+        LOAD_FUNC(alEffectf);
+        LOAD_FUNC(alEffectfv);
+        LOAD_FUNC(alGetEffecti);
+        LOAD_FUNC(alGetEffectiv);
+        LOAD_FUNC(alGetEffectf);
+        LOAD_FUNC(alGetEffectfv);
+
+        LOAD_FUNC(alGenFilters);
+        LOAD_FUNC(alDeleteFilters);
+        LOAD_FUNC(alIsFilter);
+        LOAD_FUNC(alFilteri);
+        LOAD_FUNC(alFilteriv);
+        LOAD_FUNC(alFilterf);
+        LOAD_FUNC(alFilterfv);
+        LOAD_FUNC(alGetFilteri);
+        LOAD_FUNC(alGetFilteriv);
+        LOAD_FUNC(alGetFilterf);
+        LOAD_FUNC(alGetFilterfv);
+
+        LOAD_FUNC(alGenAuxiliaryEffectSlots);
+        LOAD_FUNC(alDeleteAuxiliaryEffectSlots);
+        LOAD_FUNC(alIsAuxiliaryEffectSlot);
+        LOAD_FUNC(alAuxiliaryEffectSloti);
+        LOAD_FUNC(alAuxiliaryEffectSlotiv);
+        LOAD_FUNC(alAuxiliaryEffectSlotf);
+        LOAD_FUNC(alAuxiliaryEffectSlotfv);
+        LOAD_FUNC(alGetAuxiliaryEffectSloti);
+        LOAD_FUNC(alGetAuxiliaryEffectSlotiv);
+        LOAD_FUNC(alGetAuxiliaryEffectSlotf);
+        LOAD_FUNC(alGetAuxiliaryEffectSlotfv);
+        if(getALError() == AL_NO_ERROR)
+        {
+            ALuint envReverb;
+            alGenEffects(1, &envReverb);
+            if(getALError() == AL_NO_ERROR)
+            {
+                alEffecti(envReverb, AL_EFFECT_TYPE, AL_EFFECT_EAXREVERB);
+                if(alGetError() == AL_NO_ERROR)
+                    DPrintf("  EAX Reverb found\n");
+                alEffecti(envReverb, AL_EFFECT_TYPE, AL_EFFECT_REVERB);
+                if(alGetError() == AL_NO_ERROR)
+                    DPrintf("  Standard Reverb found\n");
+
+                alDeleteEffects(1, &envReverb);
+                getALError();
+            }
+
+            alGenAuxiliaryEffectSlots(1, &EnvSlot);
+            alGenFilters(2, EnvFilters);
+            if(getALError() == AL_NO_ERROR)
+            {
+                alFilteri(EnvFilters[0], AL_FILTER_TYPE, AL_FILTER_LOWPASS);
+                alFilteri(EnvFilters[1], AL_FILTER_TYPE, AL_FILTER_LOWPASS);
+                if(getALError() == AL_NO_ERROR)
+                    DPrintf("  Lowpass found\n");
+                else
+                {
+                    alDeleteFilters(2, EnvFilters);
+                    EnvFilters[0] = EnvFilters[1] = 0;
+                    alDeleteAuxiliaryEffectSlots(1, &EnvSlot);
+                    EnvSlot = 0;
+                    getALError();
+                }
+            }
+            else
+            {
+                alDeleteFilters(2, EnvFilters);
+                alDeleteAuxiliaryEffectSlots(1, &EnvSlot);
+                EnvFilters[0] = EnvFilters[1] = 0;
+                EnvSlot = 0;
+                getALError();
+            }
+        }
+    }
+
+    if(EnvSlot)
+        Printf("  EFX enabled\n");
+}
```

Comment:
Wow. We added so much stuff? What is all of this? :)

Reply:
I think I tried creating a new constructor since I didn't want to break other calls. Didn't overload :p

Reply:
Please don't do this. You can move shared code to a new function, like `init()` and call it from both ctor overloads.